dash/dash.py

```diff
@@ -558,11 +668,56 @@ def dispatch(self):
         return self.callback_map[target_id]['callback'](*args)
 
     def _setup_server(self):
+        if self.config.include_assets_files:
+            self._walk_assets_directory()
+
         self._generate_scripts_html()
         self._generate_css_dist_html()
 
+    def _walk_assets_directory(self):
+        walk_dir = self.config.assets_folder
+        slash_splitter = re.compile(r'[\\/]+')
+
+        def add_resource(p):
+            res = {'asset_path': p}
+            if self.config.assets_external_path:
+                res['external_url'] = '{}{}'.format(
+                    self.config.assets_external_path, path)
+            return res
+
+        for current, _, files in os.walk(walk_dir):
+            if current == walk_dir:
+                base = ''
+            else:
+                s = current.replace(walk_dir, '').lstrip('\\').lstrip('/')
+                splitted = slash_splitter.split(s)
+                if len(splitted) > 1:
+                    base = '/'.join(slash_splitter.split(s))
+                else:
+                    base = splitted[0]
+
+            for f in files:
+                if base:
+                    path = '/'.join([base, f])
+                else:
+                    path = f
+
+                if f.endswith('js'):
+                    self.scripts.append_script(add_resource(path))
+                elif f.endswith('css'):
+                    self.css.append_css(add_resource(path))
+                elif f == 'favicon.ico':
+                    self._favicon = path
+
+    def add_meta_tag(self, meta):
+        self._meta_tags.append(meta)
+
     def run_server(self,
                    port=8050,
                    debug=False,
                    **flask_run_options):
+        bp = flask.Blueprint('assets', 'assets',
+                             static_folder=self.config.assets_folder,
+                             static_url_path=self.config.assets_url_path)
+        self.server.register_blueprint(bp)
```

Comment:
I don't know much about Flask Blueprints, but what about users who are using a WSGI server like Gunicorn rather than the builtin server. Would this setup mean that they won't get the assets added?

Reply:
It will work, wsgi just take the flask instance as the wsgi app.

Reply:
Oh you'r right, thanks for bringing it up, I'll move it.